{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyM8+LsD7VvSeZKpbLw30Mfj"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["## Install Miniconda, actiavte conda & create environment\n","### <font color=\"brown\"> type 'y' to proceed <font>"],"metadata":{"id":"_hkD0hYuoKeR"}},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"id":"vzgby9wAn23u","executionInfo":{"status":"ok","timestamp":1756971060441,"user_tz":-300,"elapsed":94606,"user":{"displayName":"Tooba Zahra","userId":"08523264776337847242"}},"outputId":"85c37c71-cc58-4caa-8c47-90e75fecea79"},"outputs":[{"output_type":"stream","name":"stdout","text":["--2025-09-04 07:29:25--  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n","Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.191.158, 104.16.32.241, 2606:4700::6810:bf9e, ...\n","Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.191.158|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 162129736 (155M) [application/octet-stream]\n","Saving to: ‘Miniconda3-latest-Linux-x86_64.sh’\n","\n","Miniconda3-latest-L 100%[===================>] 154.62M   181MB/s    in 0.9s    \n","\n","2025-09-04 07:29:26 (181 MB/s) - ‘Miniconda3-latest-Linux-x86_64.sh’ saved [162129736/162129736]\n","\n","PREFIX=/usr/local\n","Unpacking bootstrapper...\n","Unpacking payload...\n","\n","Installing base environment...\n","\n","Preparing transaction: ...working... done\n","Executing transaction: ...working... done\n","installation finished.\n","WARNING:\n","    You currently have a PYTHONPATH environment variable set. This may cause\n","    unexpected behavior when running the Python interpreter in Miniconda3.\n","    For best results, please verify that your PYTHONPATH only points to\n","    directories of packages that are compatible with the Python interpreter\n","    in Miniconda3: /usr/local\n","no change     /usr/local/condabin/conda\n","no change     /usr/local/bin/conda\n","no change     /usr/local/bin/conda-env\n","no change     /usr/local/bin/activate\n","no change     /usr/local/bin/deactivate\n","no change     /usr/local/etc/profile.d/conda.sh\n","no change     /usr/local/etc/fish/conf.d/conda.fish\n","no change     /usr/local/shell/condabin/Conda.psm1\n","no change     /usr/local/shell/condabin/conda-hook.ps1\n","no change     /usr/local/lib/python3.13/site-packages/xontrib/conda.xsh\n","no change     /usr/local/etc/profile.d/conda.csh\n","modified      /root/.bashrc\n","\n","==> For changes to take effect, close and re-open your current shell. <==\n","\n","\u001b[1;33mJupyter detected\u001b[0m\u001b[1;33m...\u001b[0m\n","\n","CondaToSNonInteractiveError: Terms of Service have not been accepted for the following channels. Please accept or remove them before proceeding:\n","    - https://repo.anaconda.com/pkgs/main\n","    - https://repo.anaconda.com/pkgs/r\n","\n","To accept these channels' Terms of Service, run the following commands:\n","    conda tos accept --override-channels --channel https://repo.anaconda.com/pkgs/main\n","    conda tos accept --override-channels --channel https://repo.anaconda.com/pkgs/r\n","\n","For information on safely removing channels from your conda configuration,\n","please see the official documentation:\n","\n","    https://www.anaconda.com/docs/tools/working-with-conda/channels\n","\n","accepted Terms of Service for \u001b[4;94mhttps://repo.anaconda.com/pkgs/main\u001b[0m\n","accepted Terms of Service for \u001b[4;94mhttps://repo.anaconda.com/pkgs/r\u001b[0m\n","\u001b[1;33mJupyter detected\u001b[0m\u001b[1;33m...\u001b[0m\n","\u001b[1;32m2\u001b[0m\u001b[1;32m channel Terms of Service accepted\u001b[0m\n","Retrieving notices: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","Channels:\n"," - defaults\n","Platform: linux-64\n","Collecting package metadata (repodata.json): | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","Solving environment: | \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /usr/local/envs/funtools\n","\n","\n","\n","Proceed ([y]/n)? y\n","\n","\n","Downloading and Extracting Packages:\n","\n","Preparing transaction: - \b\bdone\n","Verifying transaction: | \b\bdone\n","Executing transaction: - \b\bdone\n","#\n","# To activate this environment, use\n","#\n","#     $ conda activate funtools\n","#\n","# To deactivate an active environment, use\n","#\n","#     $ conda deactivate\n","\n"]}],"source":["!wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n","!chmod +x Miniconda3-latest-Linux-x86_64.sh\n","!bash ./Miniconda3-latest-Linux-x86_64.sh -b -f -p /usr/local\n","!source /usr/local/etc/profile.d/conda.sh && conda init\n","!source /usr/local/etc/profile.d/conda.sh && conda create -n funtools\n","!source /usr/local/etc/profile.d/conda.sh && conda tos accept --override-channels --channel https://repo.anaconda.com/pkgs/main\n","!source /usr/local/etc/profile.d/conda.sh && conda tos accept --override-channels --channel https://repo.anaconda.com/pkgs/r\n","\n","!source /usr/local/etc/profile.d/conda.sh && conda create -n funtools\n","!source /usr/local/etc/profile.d/conda.sh && conda activate funtools"]},{"cell_type":"code","source":["!mkdir counts"],"metadata":{"id":"GtjaG4pQq6BI"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["%cd /content/counts"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Zh3jbfMmrCpG","executionInfo":{"status":"ok","timestamp":1756972248516,"user_tz":-300,"elapsed":24,"user":{"displayName":"Tooba Zahra","userId":"08523264776337847242"}},"outputId":"b68e6e25-f60d-4f61-f08b-e09f053c27b0"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/counts\n"]}]},{"cell_type":"markdown","source":["### Download dataset"],"metadata":{"id":"y7l3JGE9riZy"}},{"cell_type":"code","source":["!wget https://ftp.ensembl.org/pub/release-114/gff3/caenorhabditis_elegans/Caenorhabditis_elegans.WBcel235.114.gff3.gz\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"2Cbt_QyttE65","executionInfo":{"status":"ok","timestamp":1756972269588,"user_tz":-300,"elapsed":1427,"user":{"displayName":"Tooba Zahra","userId":"08523264776337847242"}},"outputId":"041f59d8-d7b1-4878-9600-8516ccbbcf6e"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["--2025-09-04 07:51:08--  https://ftp.ensembl.org/pub/release-114/gff3/caenorhabditis_elegans/Caenorhabditis_elegans.WBcel235.114.gff3.gz\n","Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n","Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.169|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 8383659 (8.0M) [application/x-gzip]\n","Saving to: ‘Caenorhabditis_elegans.WBcel235.114.gff3.gz’\n","\n","Caenorhabditis_eleg 100%[===================>]   8.00M  10.1MB/s    in 0.8s    \n","\n","2025-09-04 07:51:09 (10.1 MB/s) - ‘Caenorhabditis_elegans.WBcel235.114.gff3.gz’ saved [8383659/8383659]\n","\n"]}]},{"cell_type":"code","source":["!gunzip Caenorhabditis_elegans.WBcel235.114.gff3.gz"],"metadata":{"id":"YROb4WEXtTk5"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!mv /content/counts/Caenorhabditis_elegans.WBcel235.114.gff3 c_elegance.gff3"],"metadata":{"id":"NhzstlpTtduh"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!source /usr/local/etc/profile.d/conda.sh && conda activate funtools && conda install -c bioconda subread -y"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"9LCMkcUHtsn1","executionInfo":{"status":"ok","timestamp":1756972493497,"user_tz":-300,"elapsed":12792,"user":{"displayName":"Tooba Zahra","userId":"08523264776337847242"}},"outputId":"c2b84ff7-6d9c-4923-e045-53fd8f70df1e"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[1;33mJupyter detected\u001b[0m\u001b[1;33m...\u001b[0m\n","\u001b[1;32m2\u001b[0m\u001b[1;32m channel Terms of Service accepted\u001b[0m\n","Channels:\n"," - bioconda\n"," - defaults\n","Platform: linux-64\n","Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","Solving environment: \\ \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /usr/local/envs/funtools\n","\n","  added / updated specs:\n","    - subread\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    subread-2.0.1              |       h5bf99c6_1        22.8 MB  bioconda\n","    ------------------------------------------------------------\n","                                           Total:        22.8 MB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main \n","  _openmp_mutex      pkgs/main/linux-64::_openmp_mutex-5.1-1_gnu \n","  libgcc-ng          pkgs/main/linux-64::libgcc-ng-11.2.0-h1234567_1 \n","  libgomp            pkgs/main/linux-64::libgomp-11.2.0-h1234567_1 \n","  subread            bioconda/linux-64::subread-2.0.1-h5bf99c6_1 \n","  zlib               pkgs/main/linux-64::zlib-1.2.13-h5eee18b_1 \n","\n","\n","\n","Downloading and Extracting Packages:\n","                                                                        \n","Preparing transaction: / \b\bdone\n","Verifying transaction: \\ \b\b| \b\b/ \b\b- \b\bdone\n","Executing transaction: | \b\bdone\n"]}]},{"cell_type":"code","source":["!source /usr/local/etc/profile.d/conda.sh && conda activate funtools && featureCounts -v"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Qvl6K8mYuSin","executionInfo":{"status":"ok","timestamp":1756972672924,"user_tz":-300,"elapsed":1022,"user":{"displayName":"Tooba Zahra","userId":"08523264776337847242"}},"outputId":"cc50e43a-bf9e-44ed-9f65-2de33b63b3e7"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","featureCounts v2.0.1\n","\n"]}]},{"cell_type":"code","source":[],"metadata":{"id":"GOOZzb3yAppj"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!source /usr/local/etc/profile.d/conda.sh && conda activate funtools && featureCounts -O -t gene -g ID -a /content/counts/c_elegance.gff3 -o /content/counts_output.txt /content/IGV/fm_1_trim_Aligned.sortedByCoord.out.bam /content/IGV/fm_2_trim_Aligned.sortedByCoord.out.bam /content/IGV/fm_3_trim_Aligned.sortedByCoord.out.bam /content/IGV/m_4_trim_Aligned.sortedByCoord.out.bam /content/IGV/m_5_trim_Aligned.sortedByCoord.out.bam /content/IGV/m_6_trim_Aligned.sortedByCoord.out.bam"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"y3GkCDGAvbXf","executionInfo":{"status":"ok","timestamp":1756973209781,"user_tz":-300,"elapsed":1637,"user":{"displayName":"Tooba Zahra","userId":"08523264776337847242"}},"outputId":"8c31d8fd-fe64-4d0e-f1d3-2a9c48d5b2ab"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n","       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n","       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n","       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n","       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n","       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n","\t  v2.0.1\n","\n","//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n","||  \u001b[0m                                                                          ||\n","||             Input files : \u001b[36m6 BAM files  \u001b[0m \u001b[0m                                   ||\n","||                           \u001b[32mo\u001b[36m fm_1_trim_Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m       ||\n","||                           \u001b[32mo\u001b[36m fm_2_trim_Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m       ||\n","||                           \u001b[32mo\u001b[36m fm_3_trim_Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m       ||\n","||                           \u001b[32mo\u001b[36m m_4_trim_Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m        ||\n","||                           \u001b[32mo\u001b[36m m_5_trim_Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m        ||\n","||                           \u001b[32mo\u001b[36m m_6_trim_Aligned.sortedByCoord.out.bam\u001b[0m \u001b[0m        ||\n","||  \u001b[0m                                                                          ||\n","||             Output file : \u001b[36mcounts_output.txt\u001b[0m \u001b[0m                               ||\n","||                 Summary : \u001b[36mcounts_output.txt.summary\u001b[0m \u001b[0m                       ||\n","||              Annotation : \u001b[36mc_elegance.gff3 (GTF)\u001b[0m \u001b[0m                           ||\n","||      Dir for temp files : \u001b[36m/content\u001b[0m \u001b[0m                                        ||\n","||  \u001b[0m                                                                          ||\n","||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n","||              Paired-end : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n","||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n","|| Multi-overlapping reads : \u001b[36mcounted\u001b[0m \u001b[0m                                         ||\n","||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n","//=================================\u001b[36m Running \u001b[0m==================================\\\\\n","||  \u001b[0m                                                                          ||\n","|| Load annotation file c_elegance.gff3 \u001b[0m... \u001b[0m                                  ||\n","||    Features : \u001b[36m19985\u001b[0m \u001b[0m                                                       ||\n","||    Meta-features : \u001b[36m19985\u001b[0m \u001b[0m                                                  ||\n","||    Chromosomes/contigs : \u001b[36m7\u001b[0m \u001b[0m                                                ||\n","||  \u001b[0m                                                                          ||\n","|| Process BAM file fm_1_trim_Aligned.sortedByCoord.out.bam... \u001b[0m               ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m54290\u001b[0m \u001b[0m                                               ||\n","||    Successfully assigned alignments : \u001b[36m45381 (83.6%)\u001b[0m \u001b[0m                       ||\n","||    Running time : \u001b[36m0.00 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Process BAM file fm_2_trim_Aligned.sortedByCoord.out.bam... \u001b[0m               ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m51821\u001b[0m \u001b[0m                                               ||\n","||    Successfully assigned alignments : \u001b[36m43648 (84.2%)\u001b[0m \u001b[0m                       ||\n","||    Running time : \u001b[36m0.00 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Process BAM file fm_3_trim_Aligned.sortedByCoord.out.bam... \u001b[0m               ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m49605\u001b[0m \u001b[0m                                               ||\n","||    Successfully assigned alignments : \u001b[36m41497 (83.7%)\u001b[0m \u001b[0m                       ||\n","||    Running time : \u001b[36m0.00 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Process BAM file m_4_trim_Aligned.sortedByCoord.out.bam... \u001b[0m                ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m55067\u001b[0m \u001b[0m                                               ||\n","||    Successfully assigned alignments : \u001b[36m42363 (76.9%)\u001b[0m \u001b[0m                       ||\n","||    Running time : \u001b[36m0.00 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Process BAM file m_5_trim_Aligned.sortedByCoord.out.bam... \u001b[0m                ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m53884\u001b[0m \u001b[0m                                               ||\n","||    Successfully assigned alignments : \u001b[36m41627 (77.3%)\u001b[0m \u001b[0m                       ||\n","||    Running time : \u001b[36m0.00 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Process BAM file m_6_trim_Aligned.sortedByCoord.out.bam... \u001b[0m                ||\n","||    Single-end reads are included. \u001b[0m                                         ||\n","||    Total alignments : \u001b[36m51962\u001b[0m \u001b[0m                                               ||\n","||    Successfully assigned alignments : \u001b[36m39653 (76.3%)\u001b[0m \u001b[0m                       ||\n","||    Running time : \u001b[36m0.00 minutes\u001b[0m \u001b[0m                                            ||\n","||  \u001b[0m                                                                          ||\n","|| Write the final count table. \u001b[0m                                              ||\n","|| Write the read assignment summary. \u001b[0m                                        ||\n","||  \u001b[0m                                                                          ||\n","|| Summary of counting results can be found in file \"/content/counts_output. \u001b[0m ||\n","|| txt.summary\" \u001b[0m                                                              ||\n","||  \u001b[0m                                                                          ||\n","\\\\============================================================================//\n","\n"]}]},{"cell_type":"code","source":["!head /content/counts_output.txt"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"WYjWEb6RvhI3","executionInfo":{"status":"ok","timestamp":1756973372941,"user_tz":-300,"elapsed":123,"user":{"displayName":"Tooba Zahra","userId":"08523264776337847242"}},"outputId":"ab0edf78-079a-472b-b9f7-b68b0efc8d96"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["# Program:featureCounts v2.0.1; Command:\"featureCounts\" \"-O\" \"-t\" \"gene\" \"-g\" \"ID\" \"-a\" \"/content/counts/c_elegance.gff3\" \"-o\" \"/content/counts_output.txt\" \"/content/IGV/fm_1_trim_Aligned.sortedByCoord.out.bam\" \"/content/IGV/fm_2_trim_Aligned.sortedByCoord.out.bam\" \"/content/IGV/fm_3_trim_Aligned.sortedByCoord.out.bam\" \"/content/IGV/m_4_trim_Aligned.sortedByCoord.out.bam\" \"/content/IGV/m_5_trim_Aligned.sortedByCoord.out.bam\" \"/content/IGV/m_6_trim_Aligned.sortedByCoord.out.bam\" \n","Geneid\tChr\tStart\tEnd\tStrand\tLength\t/content/IGV/fm_1_trim_Aligned.sortedByCoord.out.bam\t/content/IGV/fm_2_trim_Aligned.sortedByCoord.out.bam\t/content/IGV/fm_3_trim_Aligned.sortedByCoord.out.bam\t/content/IGV/m_4_trim_Aligned.sortedByCoord.out.bam\t/content/IGV/m_5_trim_Aligned.sortedByCoord.out.bam\t/content/IGV/m_6_trim_Aligned.sortedByCoord.out.bam\n","gene:WBGene00022277\tI\t4116\t10230\t-\t6115\t1\t1\t0\t1\t1\t1\n","gene:WBGene00022276\tI\t11495\t16837\t+\t5343\t5\t3\t1\t5\t6\t7\n","gene:WBGene00022278\tI\t17484\t26781\t-\t9298\t6\t7\t3\t5\t1\t1\n","gene:WBGene00022279\tI\t27589\t33150\t-\t5562\t0\t1\t2\t7\t2\t1\n","gene:WBGene00022275\tI\t43733\t44677\t+\t945\t0\t0\t0\t0\t0\t0\n","gene:WBGene00044345\tI\t47461\t49860\t+\t2400\t3\t0\t2\t2\t0\t0\n","gene:WBGene00021677\tI\t49921\t54426\t+\t4506\t3\t0\t3\t3\t2\t0\n","gene:WBGene00021678\tI\t55293\t64066\t-\t8774\t0\t0\t0\t1\t1\t1\n"]}]},{"cell_type":"code","source":[],"metadata":{"id":"-3Eh_v5GxSRf"},"execution_count":null,"outputs":[]}]}